src/hooks/use-pull-push-timestamps.ts

```diff
@@ -1,92 +1,97 @@
 import { __, sprintf } from '@wordpress/i18n';
 import { formatDistanceToNow } from 'date-fns';
-import { useCallback, useState, useEffect } from 'react';
+import { useCallback, useState } from 'react';
+import { getIpcApi } from '../lib/get-ipc-api';
+import { SyncSite } from './use-fetch-wpcom-sites';
 
-const SYNC_TIMESTAMPS_STORAGE_KEY = 'wp-studio-sync-timestamps';
+export function usePullPushTimestamps() {
+	const [ tooltips, setTooltips ] = useState< Record< string, string > >( {} );
 
-interface SyncTimestamps {
-	[ localSiteId: string ]: {
-		[ connectedSiteId: number ]: {
-			pull?: number;
-			push?: number;
-		};
+	const fetchConnectedSite = async ( siteId: string, connectedSiteId: number ) => {
+		const connectedSites = await getIpcApi().getConnectedWpcomSites( siteId );
+		return connectedSites.find( ( site ) => site.id === connectedSiteId );
 	};
-}
 
-export function usePullPushTimestamps() {
-	const [ timestamps, setTimestamps ] = useState< SyncTimestamps >( () => {
-		try {
-			const stored = localStorage.getItem( SYNC_TIMESTAMPS_STORAGE_KEY );
-			return stored ? JSON.parse( stored ) : {};
-		} catch ( e ) {
-			console.error( 'Failed to parse sync timestamps:', e );
-			return {};
-		}
-	} );
+	const getLastSyncTimeWithType = useCallback(
+		async ( siteId: string, connectedSiteId: number, type: 'pull' | 'push' ): Promise< string > => {
+			try {
+				const site = await fetchConnectedSite( siteId, connectedSiteId );
+				if ( ! site ) return __( 'Site not found.' );
 
-	useEffect( () => {
-		try {
-			localStorage.setItem( SYNC_TIMESTAMPS_STORAGE_KEY, JSON.stringify( timestamps ) );
-		} catch ( e ) {
-			console.error( 'Failed to save sync timestamps:', e );
-		}
-	}, [ timestamps ] );
+				const timestamp = type === 'pull' ? site.lastPullTimestamp : site.lastPushTimestamp;
 
-	const updateTimestamp = useCallback(
-		( localSiteId: string, connectedSiteId: number, type: 'pull' | 'push' ) => {
-			setTimestamps( ( prev ) => {
-				const newTimestamps = { ...prev };
-				if ( ! newTimestamps[ localSiteId ] ) {
-					newTimestamps[ localSiteId ] = {};
-				}
-				if ( ! newTimestamps[ localSiteId ][ connectedSiteId ] ) {
-					newTimestamps[ localSiteId ][ connectedSiteId ] = {};
+				if ( timestamp === null ) {
+					return type === 'pull'
+						? __( 'You have not pulled this site yet.' )
+						: __( 'You have not pushed this site yet.' );
 				}
-				newTimestamps[ localSiteId ][ connectedSiteId ][ type ] = Date.now();
-				return newTimestamps;
-			} );
+
+				return sprintf(
+					type === 'pull'
+						? __( 'You pulled this site %s ago.' )
+						: __( 'You pushed this site %s ago.' ),
+					formatDistanceToNow( new Date( timestamp ) )
```

Comment:
I think we need to pass the current locale, to translate the time difference to the user's language.
You can grab it from `const { locale } = useI18nData();`

Reply:
I see, I think this issue is already present outside of this PR. I tried passing just the `locale` but it did not work well for translating dates. I added a small hook and used `formatDistance` function that we already had. This is what it looks like now:

<img width="987" alt="Screenshot 2024-12-04 at 10 32 38 AM" src="https://github.com/user-attachments/assets/e55237d0-b0ff-4e94-9720-5a3637e021c4">

To test this, I added the following line:

```
            "You pulled this site %s ago.": [
                "Pobrałeś tę witrynę %s temu."
            ]
```
to the `studio-pl.jed.json` and loaded the app in Polish. Let me know what you think!